plugins/inventory/foreman.py

```diff
@@ -83,8 +83,66 @@
         type: string
       batch_size:
         description: Number of hosts per batch that will be retrieved from the Foreman API per individual call
-        type: integer
+        type: int
         default: 250
+      foreman:
+        description: foreman server related configuration
+        type: dict
+        use_reports_api:
+            description: Use Reporting API.
+            type: boolean
+            default: False
+      report:
+        description: report API specific configuration
+        type: dict
+        poll_interval:
+            description: The polling interval between 2 calls to the report_data endpoint while polling.
+            type: int
+            default: 10
+        want_organization:
+            description: Toggle, if true the inventory will fetch organization the host belongs to and create groupings for the same.
+            type: boolean
+            default: True
+        want_location:
+            description: Toggle, if true the inventory will fetch location the host belongs to and create groupings for the same.
+            type: boolean
+            default: True
+        want_ipv4:
+            description: Toggle, if true the inventory will fetch ipv4 address of the host.
+            type: boolean
+            default: True
+        want_ipv6:
+            description: Toggle, if true the inventory will fetch ipv6 address of the host.
+            type: boolean
+            default: True
+        want_host_group:
+            description: Toggle, if true the inventory will fetch host_groups and create groupings for the same.
+            type: boolean
+            default: True
+        want_subnet:
+            description: Toggle, if true the inventory will fetch subnet.
+            type: boolean
+            default: True
+        want_subnet_v6:
+            description:  Toggle, if true the inventory will fetch ipv6 subnet.
+            type: boolean
+            default: True
+        want_smart_proxies:
+            description: Toggle, if true the inventory will fetch smart proxy that the host is registered to.
+            type: boolean
+            default: True
+        want_content_facet_attributes:
+            description: Toggle, if true the inventory will fetch content view details that the host is tied to.
+            type: boolean
+            default: True
+        want_host_params:
+            description: Toggle, if true the inventory will fetch host parameters.
+            type: boolean
+            default: True
```

Comment:
How does that compare to the `want_params` option we have outside the `report` section?

Reply:
Serves the same purpose except this one is more report specific as in sent to the report API..

Reply:
Right, but isn't it weird (from a user PoV) to have to set it again, just because a different backend is used?

Reply:
For facts, you also do use the global one: `self.want_facts = self.get_option('want_facts')`